test/testlib/multicluster_utilities.go

```diff
@@ -243,3 +253,97 @@ func AdjustPodTimeout(podName string, timeout time.Duration) time.Duration {
 	}
 	return timeout
 }
+
+const (
+	DNS_CONFIG_MARKER = "# BEGIN BASE CONFIG"
+	COREDNS_NS        = "kube-system"
+	COREDNS_CM        = "coredns"
+	COREFILE_KEY      = "Corefile"
+	DNS_SERVICE       = "dns-service"
+)
+
+func getClusterDnsServer(t *testing.T, clusterContext context.Context) string {
+	var ret string
+	DeployWithContext(t,
```

Comment:
Nit: It's better to use `WithClusterDeployment` as this is just executing `kubectl`.

Reply:
The difference is in the callback used, which does not take a `helm.Options`?

I do use the `helm.Options` to get the `k8s.KubectlOptions` that I invoke `kubectl`, but maybe there is another way to do it. Either way, I'd rather not go through another round of manual testing, so I think I'll leave it the way it is.